MsBuild.NuGet.Pack/MergeNuSpecTask.cs

```diff
@@ -175,6 +175,22 @@ private static IEnumerable<XElement> GetProjectPackages(string packageConfig)
         }
 
         /// <summary>
+        ///     Gets a regex that can be used to check package exclusions.
+        /// </summary>
+        /// <param name="exclusionPatterns">
+        ///     The package exclusion patterns.
+        /// </param>
+        /// <returns>
+        ///     A <see cref="Regex" />.
+        /// </returns>
+        public static Regex GetPackageExclusions(string exclusionPatterns)
+        {
+            var regexPatterns = exclusionPatterns.Split(';').Select(x => Regex.Escape(x).Replace(@"\*", ".*").Replace(@"\?", "."));
+
+            return new Regex("^" + String.Join("|", regexPatterns) + "$");
```

Comment:
The start and end constraints are not applied against each |. For example, this expression might result in ```^First|Second$``` for which ```FirstStuff``` would match, but I don't think that is your intention.

I think this line should be:

```
return new Regex("^(" + String.Join("|", regexPatterns) + ")$");
```

This groups the OR such that the ^ and $ are applied against the group rather than the boundary entries.